fgpyo/sam/__init__.py

```diff
@@ -606,6 +607,110 @@ def __getitem__(
         return self.elements[index]
 
 
+@enum.unique
+class PairOrientation(enum.Enum):
+    """Enumerations of read pair orientations."""
+
+    FR = "FR"
+    """A pair orientation for forward-reverse reads ("innie")."""
+
+    RF = "RF"
+    """A pair orientation for reverse-forward reads ("outie")."""
+
+    TANDEM = "TANDEM"
+    """A pair orientation for tandem (forward-forward or reverse-reverse) reads."""
+
+    @classmethod
+    def build(
+        cls, r1: AlignedSegment, r2: Optional[AlignedSegment] = None
```

Comment:
I see this pattern of `r1` and optional `r2` in several places.  I think it would be better to use different naming (e.g. `read` and `mate`) - because I don't think you actually care that the first read is `first_of_pair`, and I think this function would work called with a single `second_of_pair` and no mate.

Reply:
I agree!

Reply:
https://github.com/fulcrumgenomics/fgpyo/pull/201/commits/6f3c514ed48c8cbd6dc4f69eafa50b06f871201c